packages/core-data/src/resolvers.js

```diff
@@ -145,7 +145,10 @@ export function* canUser( action, resource, id ) {
 		throw new Error( `'${ action }' is not a valid action.` );
 	}
 
-	const path = id ? `/wp/v2/${ resource }/${ id }` : `/wp/v2/${ resource }`;
+	const apiPrefix = resource && resource.isExperimental ? '/__experimental/' : '/wp/v2/';
```

Comment:
I'm not so sure about the interface we've created for this.

A few thoughts occur to me:

- Depending on how we define the endpoint, can `__experimental` be part of the `resource` ?
   - i.e. `/wp/v2/__experimental/media`
- Could the function signature allow one to customize the namespace altogether, vs. "experimental" as being a specialized concept?
   - i.e. `{ endpoint: 'media', namespace: '/wp/__experimental' }`
   - Aside: Is "endpoint" the correct verbiage here?
- Is there any potential reuse or conflict from the changes or ideas proposed at #18956 ?

Reply:
I can't speak to the history of how `__experimental`, which appears to be used in 3 different `WP_REST_Controller` extensions (`widget-forms`, `block-directory` & `widget-areas`) came to pass and the reasoning for not having them be part of the resource. I wouldn't be opposed to having to it but I don't know who would need to be looped in for context.